ibis/tests/expr/test_struct.py

```diff
@@ -42,3 +55,34 @@ def test_struct_pickle():
     )
 
     assert_pickle_roundtrip(struct_scalar_expr)
+
+
+def test_unpack(t):
+    assert t.a.unpack().equals(t[_.a.b, _.a.c, _.d])
+
+
+def test_unpack_from_table(t):
+    assert t.unpack("a").equals(t[_.a.b, _.a.c, _.d])
+
+
+def test_unpack_join(t, s):
+    join = t.join(s, t.d == s.a.g)
+    result = join.a_y.unpack()
+    expected = join[_.a_x, _.d, _.a_y.f, _.a_y.g]
+    assert result.equals(expected)
+
+
+def test_unpack_join_from_table(t, s):
+    join = t.join(s, t.d == s.a.g)
+    result = join.unpack("a_y")
+    expected = join[_.a_x, _.d, _.a_y.f, _.a_y.g]
+    assert result.equals(expected)
+
+
+def test_nested_unpack():
+    t = ibis.table(
+        {"a": "struct<b:struct<x: int, y: int>, c: string>", "d": "string"},
+        name="t",
+    )
+    expr = t.a.b.unpack()
+    assert expr.columns == list("xycd")
```

Comment:
Wait, I find this change _super_ confusing - I would've expected `list("xyd")`.

I read `obj.unpack()` as taking all the immediate fields of `obj` and unpacking them into the parent table. Since `c` isn't in `obj` (but rather a parent struct of `obj`), I find it very confusing that it's included in the unpacked columns. What if the column names conflict (say `c` is named `x` instead)?

Reply:
Hm, yeah that's a good point. I'm not sure I like this change either. It *is* very convenient but incredibly non-obvious and hard to predict.

I think having to explicitly `.unpack()` is better. I'll revisit.

Reply:
Tinkering with this some more I think there are only two things that `t.a.b.unpack()` (t.unpack("a.b") would be no different) could mean:

1. Recursively unpack until the child is no longer a `StructField` operation.
2. Fail, because we want people to be explicit about the operation.

In any case, unpacked struct field names that overlap with existing child table field names are an error because duplicate column names are not allowed. This is true whether or not we take care of unpacking the entire dotted path for the user.

Reply:
Ok, I think was confused by why you were expecting `xyd` here, but now I see understand. It seems like there is a third option for behavior which is inlining the struct without unpacking the children.

Reply:
Alright, I've implemented this in a way that I think reflects a reasonable compromise between convenience and semantic clarity. `t.a.b.unpack()` works, but doesn't recursively unpack, it pulls up the nested struct into top level fields.

Reply:
Yeah, that's what I was asking for (inline, but not recursively). Sorry that wasn't clear above.